controllers/keywordController.js

```diff
@@ -0,0 +1,49 @@
+const keywordModel = require("../models/keywordModel");
+const groupModel = require("../models/groupModel");
+
+const isValid = (data) => {
+  return data !== null && data !== undefined;
+};
+
+const create = async (req, res) => {
+  if (!isValid(req.body.groupName) || !isValid(req.body.keyword) || !isValid(req.body.ownerId)) {
+    res.status(400).send({ message: "[Error: keywordCreate] Content can not be empty!" });
+  }
+
+  const isDuplicatedKeyword =
+    keywordModel.find({ groupId: req.body.groupId, keyword: req.body.keyword }) === null;
+
+  if (isDuplicatedKeyword) {
+    res.status(400).send({ message: "[Error: keywordCreate] Content can not be duplicated!" });
+  }
+
+  const { groupId, groupName, keyword, ownerId } = req.body;
+
+  try {
+    if (groupId === null) {
+      const resultGroupCreated = await groupModel.create({
+        name: groupName,
+        ownerId,
+        keywordList: [keyword],
+      });
+      await keywordModel.create({
+        keyword,
+        ownerId: ownerId,
+      });
+      res.status(201).json(resultGroupCreated);
+    } else {
+      await keywordModel.create({ keyword, ownerId });
+      const queryGroupUpdated = await groupModel.findByIdAndUpdate(
+        groupId,
+        { $push: { keywordList: keyword } },
+        { new: true }
+      );
+      res.status(201).json(queryGroupUpdated);
```

Comment:
```suggestion
    await keywordModel.create({ keyword, ownerId });
    
    if (groupId === null) {
      const resultGroupCreated = await groupModel.create({
        name: groupName,
        ownerId,
        keywordList: [keyword],
      });
      
      res.status(201).json(resultGroupCreated);
    } else {
      const queryGroupUpdated = await groupModel.findByIdAndUpdate(
        groupId,
        { $push: { keywordList: keyword } },
        { new: true }
      );
      
      res.status(201).json(queryGroupUpdated);
```
저희가 논의를 통해서 group의 Schema 구조를 `keywordList` 에서 `keywordIdList`로 변경하기로 했기 때문에,
키워드를 생성하는 로직인 `keywordModel.create` 부분을 최상단으로 위치시키고, 그 이후에 groupId 유무에 따른 조건을 처리하면 좋을 것 같습니다!

Reply:
제가 제안드린 방법에 대해 더 생각해보니 groupId가 `null`이 아닌 경우엔 groupId의 유무 확인이 키워드 생성 이후에 진행되고 있어요.
그러면 groupId가 `null`은 아니지만 유효한 id가 아니라면 키워드만 생성되고 에러가 발생할 경우가 있을 것 같네요..!

해당 함수 최상단의 유효성 검사하는 로직에서
`const isExistGroupId = groupModel.find({ groupId: req.body.groupId }) === null;`
해당 코드를 추가하고, 
하단의 else 블록에선 `findByIdAndUpdate` 쿼리를 `updateOne` 쿼리로 변경하면 될 것 같습니다.

이러한 방법도 있다는 라는 걸 전달드려요! 🙇‍♀️

Reply:
감사합니다, 한솔님! 공통 로직이 포함되어 있어서 밖으로 빼두는 것이 좋은 방향 같습니다.
구두로 논의한 부분 바탕으로 아래 처럼 코드 수정했습니다.

수정한 코드로 테스트도 완료해서 반영하도록 하겠습니다. :)

```js
// `!==null`로 변경
const isValidGroupId = groupModel.find({ groupId: req.body.groupId }) !== null;

const { groupId, groupName, keyword, ownerId } = req.body;

try {
// `create`의 결과 생성된 `keywordId`를 추가하기 위해 `_id` 접근하는 로직 추가
  const resultKeywordCreated = await keywordModel.create({ keyword, ownerId });
  const { _id: keywordIdCreated } = resultKeywordCreated;

  if (groupId === null) {
    const resultGroupCreated = await groupModel.create({
      name: groupName,
      ownerId,
      keywordIdList: [keywordIdCreated],
    });
    res.status(201).json(resultGroupCreated);
// `groupId`가 유효한 경우만 정상 처리하기 위해 `else if (isValidGroupId)`로 변경
  } else if (isValidGroupId) {
// `updateOne` 적용시 update 결과를 반환하지 못해 기존 처럼 `findByIdAndUpdate` 적용.
    const queryGroupUpdated = await groupModel.findByIdAndUpdate(
      { _id: groupId },
      { $push: { keywordIdList: keywordIdCreated } },
      { new: true }
    );
    res.status(201).json(queryGroupUpdated);
}
```